internal/db/storage/blob/s3/s3_test.go

```diff
@@ -50,42 +60,136 @@ func TestNew(t *testing.T) {
 		if !errors.Is(err, w.err) {
 			return errors.Errorf("got_error: \"%#v\",\n\t\t\t\twant: \"%#v\"", err, w.err)
 		}
-		if !reflect.DeepEqual(got, w.want) {
-			return errors.Errorf("got: \"%#v\",\n\t\t\t\twant: \"%#v\"", got, w.want)
+
+		opts := []comparator.Option{
+			comparator.AllowUnexported(client{}),
+			comparator.Comparer(func(want, got errgroup.Group) bool {
+				return reflect.DeepEqual(want, got)
+			}),
+			comparator.Comparer(func(want, got *s3.S3) bool {
+				return want != nil && got != nil
+			}),
+			comparator.Comparer(func(want, got *session.Session) bool {
+				return reflect.DeepEqual(want, got)
+			}),
+			comparator.Comparer(func(want, got func(string) (reader.Reader, error)) bool {
+				return want != nil && got != nil
```

Comment:
should we comparator this object more accurately ?

Reply:
I think we need not comparator this object more accurately.
we can not comparator `got` and `want` more accurately because reader initialize function (readerFunc) will be set from the unexported default function. so we can not get pointer.

Reply:
I am not sure but the test code are using mock and it should be fine 🤔